### PR TITLE
:sparkles: Add Dev Community Presence

### LINKED ISSUE #3352: DEV Community | dev.to
**Prerequisites and essential questions** 
- [x] Is it a popular site?
- [x] Is the website older than 2 months? 
- [ ] Is the site locked to a specific country/region?
- [ ] Is the site a paid service? (e.g. Netflix, Hulu)
- [ ] Does the website feature NSFW content? (e.g. porn, etc...)
- [ ] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**What's your Discord username?** 
rippledwater#9411
**What's the name of the service?** 
dev.to | DEV Community
**What should the Presence display?** 
Homepage, search page, profile page, post, podcast
**If possible, please provide a logo for the service (512x512 minimum)** 
dev-black

### PR DESCRIPTION
**Pull request by TheGamerzs**
Resolves #3352
image
image
image

### CODE REVIEW

File: websites/D/Dev Community/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    "en": "A constructive and inclusive social network for software developers. With you every step of your journey."
+  },
+  "url": ["dev.to", "shop.dev.to"],
+  "version": "1.0.0",
+  "logo": "https://i.imgur.com/dcx0aN7.png",
+  "thumbnail": "https://i.imgur.com/1qV4crv.png",
+  "color": "#7289DA",
+  "tags": ["tutorials", "dev-to", "community"],
+  "category": "socials",
+  "buttons": true,
```
Comment (EncryptedDev) [inline websites/D/Dev Community/dist/metadata.json:17]: not a thing?

Comment (TheGamerzs) [reply websites/D/Dev Community/dist/metadata.json:17]: Will update

**Review by StrikerFRFX — CHANGES_REQUESTED** [review]
Hi @TheGamerzs, 
I've requested changes below in my review which you should take a look at 😄 
 -  In addition to the changes requested, please change the folder name and such to match the service name of `DEV Community`

Thanks,
**Striker**, PreMiD Reviewer Team.
Contact: me@strikerfrfx.dev, Striker#1337

File: websites/D/Dev Community/dist/metadata.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (TheGamerzs) [reply websites/D/Dev Community/dist/metadata.json:17]: Read prevoius conversation

File: websites/D/DEV Community/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
+  "author": {
+    "name": "The Gamerzs",
+    "id": "730142125181894657"
+  },
+  "service": "Dev Community",
```
Comment (StrikerFRFX) [inline websites/D/DEV Community/dist/metadata.json:6]: Service name still not changed ._.

**Review by StrikerFRFX — CHANGES_REQUESTED** [review]
Hi @TheGamerzs, 
I've requested changes below in my review which you should take a look at 😄 
 -  In addition to the service name being changed, pleaee make sure you change the  Discord App name and post a screenshot as proof, thanks 👍 

Thanks,
**Striker**, PreMiD Reviewer Team.
Contact: me@strikerfrfx.dev, Striker#1337

File: websites/D/DEV Community/presence.ts [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+        document.querySelector("video")
+      )[1];
+      if (contentStateKey == "play" && endTimestamp > 0) {
+        presenceData.endTimestamp = endTimestamp;
+      } else {
+        delete presenceData.startTimestamp;
+        delete presenceData.endTimestamp;
+      }
+      break;
+      break;
```
Comment (ririxidev) [inline websites/D/DEV Community/presence.ts:171]:
```suggestion
      if (contentStateKey === "play" && endTimestamp > 0)
        presenceData.endTimestamp = endTimestamp;
      else
        delete presenceData.endTimestamp;
      break;
```

File: websites/D/DEV Community/presence.ts [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+
+      endTimestamp = presence.getTimestampsfromMedia(
+        document.querySelector("#audio")
+      )[1];
+      if (contentStateKey == "play" && endTimestamp > 0) {
+        presenceData.endTimestamp = endTimestamp;
+      } else {
+        delete presenceData.startTimestamp;
+        delete presenceData.endTimestamp;
+      }
```
Comment (ririxidev) [inline websites/D/DEV Community/presence.ts:204]:
```suggestion
      if (contentStateKey === "play" && endTimestamp > 0)
        presenceData.endTimestamp = endTimestamp;
      else
        delete presenceData.endTimestamp;
```

**Review by ririxidev — CHANGES_REQUESTED** [review]
Heya @TheGamerzs!
You should check out thing(s) that I requested to change in my review. Please look at them!

Regards,
**ririxi**. PreMiD Reviewer Team.
Contact: mail@ririxi.dev, ririxi#2721.

**Comment by TheGamerzs** [comment]
@StrikerFRFX 
image

**Comment by TheGamerzs** [comment]
Fixed
image

**Review by ririxidev — APPROVED** [review]
ok to me, check if it compiles too

**Review by Bas950 — APPROVED** [review]
LGTM

**Comment by dalhaqq** [comment]
Woah, nice, thanks!

**Pull request merged**